### DISCUSSION TITLE
Can I set Sprit tint as gradient?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7666] [Q&A]
I read this post https://github.com/pixijs/pixijs/issues/2216
It would be really cool feature.
Is there a way to do this?

**Comment by ivanpopelyshev** [comment]
Make a mesh-shader https://pixijs.io/examples/#/mesh-and-shaders/triangle-color.js apply to quad https://pixijs.io/examples/#/mesh-and-shaders/shadertoy-mesh.js 

alternatively, 
1. make sprite with custom `calculateVertices` that also puts those colors in array, like here: https://github.com/pixijs/pixi-heaven/blob/eb554499afe3fcbb42bb4d4f77edb2ee0c27b7fa/src/twotint/sprites/SpriteH.ts#L98
2. make a custom BatchRenderer that interleaves this array in vertices: https://github.com/pixijs/pixi-heaven/blob/master/src/twotint/sprites/HeavenBatchRenderer.ts  . In your case you dont even have to change geometry, just use different colors for each vertex!
3. publish it as a plugin (copy graphics-smooth or some other config, pixi-heaven maybe)
4. make examples
5. look how people use it. some people. rarely.
6. wonder why pixijs team members dont care about your "cool feature"
7. cry
8. go play civ6 for hours
9. imagine new feature, go to step 1

**Reply by ivanpopelyshev** [reply]
in your case you dont have to change geometry and shader. just add different colors in interleave, not the same one
